### PR TITLE
Add networkpool support on AWS Clusters

### PR DESCRIPTION
**Pull request by paurosello**
## Checklist

- [x] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.
- [x] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

**Comment by xh3b4sd** [comment]
Some thoughts. 

* Are there some tests we can extend or add? 
* Should we have additional validation? 
* How is the optionality of network pools still ensured?

**Comment by paurosello** [comment]
I created some tests, thanks for reminding :+1: 

I am checking that setting an empty value for the NetworkPool produces same output as not even specifying the attribute in order to test optionality.

The second test I added is to check that the CR is correctly generated when specifying a NetworkPool

File: pkg/apis/infrastructure/v1alpha2/aws_cluster_types_test.go [hunk 1]
```diff
@@ -144,3 +154,86 @@ func newAWSClusterExampleCR() *AWSCluster {
 
 	return cr
 }
+
+func newAWSClusterEmptyNetworkPoolCR() *AWSCluster {
+	awscluster := newAWSClusterExampleCR()
```
Comment (xh3b4sd) [inline pkg/apis/infrastructure/v1alpha2/aws_cluster_types_test.go:159]: Just FMI why do we have `newAWSClusterExampleCR` and `NewAWSClusterCR`? 🙈

Comment (xh3b4sd) [reply pkg/apis/infrastructure/v1alpha2/aws_cluster_types_test.go:159]: I see in https://github.com/giantswarm/apiextensions/blob/58f996a5fea29796572cf3909214cabffbd0b37c/pkg/apis/infrastructure/v1alpha2/aws_cluster_types.go#L27-L37 that the spec is empty when using `NewAWSClusterCR`. /shrug

**Review by xh3b4sd — APPROVED** [review]

File: pkg/apis/infrastructure/v1alpha2/aws_cluster_types_test.go (hunk 1 above)
Comment (paurosello) [reply pkg/apis/infrastructure/v1alpha2/aws_cluster_types_test.go:159]: yup, not sure about why do we have that empty CR function

**Pull request merged**